src/Frontend/Core/Engine/Model.php

```diff
@@ -326,6 +329,29 @@ public static function getModuleSettings($module)
     }
 
     /**
+     * Creates router for module based on routing file
+     *
+     * @param $module
+     * @return null|Router
+     */
+    public static function getModuleRouter($module)
```

Comment:
That would really be awesome. I don't have a lot of time soon, but I'll check if I can find some time to do this. This would likely fix a lot of problems and make a lot of code cleaner.